core/server/app.go

```diff
@@ -32,33 +22,21 @@ type BlockRequest[T transaction.Tx] struct {
 	IsGenesis bool
 }
 
+// BlockResponse defines the response structure for a block coming from the state transition function to consensus server.
 type BlockResponse struct {
-	Apphash          []byte
 	ValidatorUpdates []appmodulev2.ValidatorUpdate
 	PreBlockEvents   []event.Event
 	BeginBlockEvents []event.Event
 	TxResults        []TxResult
 	EndBlockEvents   []event.Event
 }
 
-type RequestInitChain struct {
-	Time          time.Time
-	ChainId       string
-	Validators    []appmodulev2.ValidatorUpdate
-	AppStateBytes []byte
-	InitialHeight int64
-}
-
-type ResponseInitChain struct {
-	Validators []appmodulev2.ValidatorUpdate
-	AppHash    []byte
-}
-
+// TxResult defines the result of a transaction execution.
 type TxResult struct {
-	Events    []event.Event
-	Resp      []transaction.Msg
-	Error     error
-	Code      uint32
+	Events    []event.Event     // Events produced by the transaction.
+	Resp      []transaction.Msg // Response messages produced by the transaction.
+	Error     error             // Error produced by the transaction.
+	Code      uint32            // Code produced by the transaction.
```

Comment:
Can we have these before the field so that they're doc comments? Also this doesn't really explain what these mean. Copilot can mostly write docs for all fields and then we can expand where needed

Reply:
i went through the useage in comets spec and how we use it in baseapp and removed fields that we previously populated with error information. Ill double check with users if thats okay